### DISCUSSION TITLE
Use Extism for client side language SDKs

### DISCUSSION THREAD

**Discussion by thepabloaguilar** [flipt-io/flipt#3590] [Ideas]
Hi, have you evaluated the [Extism](https://extism.org/) for building the SDKs with client side evaluation? I know it's not available for all the languages Flipt supports right now but maybe it can reduce some "troubles" for some languages like Go where we need to compile using cgo!

**Comment by GeorgeMac** [comment]
Extism is really cool and we love a bit of WASM! We intended to port the Go one to Wazero (same runtime as extism) but we never got around to it in the end (still an option).

Extism definitely would be a nice choice to defer many headaches to Extism itself (e.g. serializing types across the host to Rust boundary and packaging up things for the variety of different archetectures).

One thing of note is that many of the Extism SDKs are just FFI shims to a WASM runtime (most likely wasmtime / some other Rust wasm runtime). The Go one aside, I think e.g. the Java, Python etc do this.
But this does push the packaging of the thing on the other side of the FFI boundary to Extism, so that would likely remove a bunch of packaging headaches.

Maybe it is worth taking one of the languages we don't support and adding support with it (I know PHP has been asked for recently)?
That could be an interesting experiment to validate Extism for our usecase.

Certainly a non-trivial amount of work to move existing clients to Extism though.
Have to rethink all the existing boundaries and rewrite the release pipelines (though they should become simpler).

**Reply by thepabloaguilar** [reply]
Got it, Idk about PHP but I'd help to make this try with Elixir if it's not a problem!
